nav2_velocity_smoother/src/velocity_smoother.cpp

```diff
@@ -246,6 +249,24 @@ void VelocitySmoother::smootherTimer()
   // proportionally to follow the same direction, within change of velocity bounds.
   // In case eta reduces another axis out of its own limit, apply accel constraint to guarantee
   // output is within limits, even if it deviates from requested command slightly.
+
+  // Don't cross zero
+  if ((current_.linear.x < 0.0 && command_->linear.x > 0) ||
+    (current_.linear.x > 0.0 && command_->linear.x < 0))
+  {
+    command_->linear.x = 0.0;
+  }
+  if ((current_.linear.y < 0.0 && command_->linear.y > 0) ||
+    (current_.linear.y > 0.0 && command_->linear.y < 0))
+  {
+    command_->linear.y = 0.0;
+  }
+  if ((current_.angular.z < 0.0 && command_->angular.z > 0) ||
+    (current_.angular.z > 0.0 && command_->angular.z < 0))
+  {
+    command_->angular.z = 0.0;
+  }
+
```

Comment:
Just a suggestion. The two conditions within an `if` statement are equivalent: they check if the two numbers have different sign with respect to each other. Then, this can be easily simplified with
`if (std::signbit(current_.linear.x * command_->linear.x)) command_->linear.x = 0.0;`
where `std::signbit(x)` return `false` if `x < 0`, which is when one between `current_.linear.x` and `command_->linear.x` has a negative value.
This would make this section shorter and cleaner.

Reply:
Oh yes! that's a good one